### DISCUSSION TITLE
Custom recommendations plugin

### DISCUSSION THREAD

**Discussion by bradbeattie** [jellyfin/jellyfin#8811] [General]
I'm looking to augment Jellyfin with a plugin I'm happy to write.

There are some movies that are strongly related, but neither collections nor tags reveal this (e.g. Lost in Translation and Her).

What I'd like to permit is the ability to hard-code recommendation pairings between movies. I suspect I need to hook into https://github.com/jellyfin/jellyfin/blob/master/Jellyfin.Api/Controllers/MoviesController.cs#L64, but I'm new to Jellyfin plugins.

Is there a guide or better yet a preexisting example of modifying recommendations I can take a look at? I don't see any in https://jellyfin.org/docs/general/server/plugins/

**Comment by bradbeattie** [comment]
Maybe I'm best to first scaffold off of https://github.com/jellyfin/jellyfin-plugin-template and come back when I hit my roadblocks.

It's looking like I'll need to implement some kind of override around https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Data/SqliteItemRepository.cs#L2418, but unclear how one goes about doing that.

**Reply by bradbeattie** [reply]
https://emby.media/community/index.php?/topic/101629-plugin-to-control-results-displayed-in-more-like-this/ suggests this might not be feasible in a plugin with the current implementation.

**Reply by elechtma** [reply]
Have you made any progress on this? I was curious if one could use gpt API to make custom collections/playlists, or recommendations. But I don't have much programming experience.

**Comment by 20BBrown14** [comment]
Several years layer but I was also was looking to have user specific recommendations. I wrote and use https://github.com/20BBrown14/NextUp for this now

**Comment by Shadowghost** [comment]
https://github.com/jellyfin/jellyfin/pull/15970 implements this in a generic way so plugins can just implement the interface
